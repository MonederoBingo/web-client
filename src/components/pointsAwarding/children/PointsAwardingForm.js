import React from "react";
import NumberInput from "../../common/NumberInput";
import TextInput from "../../common/TextInput";

const PointsAwardingForm = (props) => {
  return (
     <div>
       <TextInput
         name="phoneNumber"
         label="Phone #"
         onChange={props.onChange}
         placeholder="Phone #"
         value={props.pointsAwarding.phoneNumber}
         />

       <NumberInput
         name="amount"
         label="Amount"
         onChange={props.onChange}
         placeholder="Amount"
         value={props.pointsAwarding.amount}
         />

       <TextInput
         name="saleKey"
         label="Sale Key"
         onChange={props.onChange}
         placeholder=""
         value={props.pointsAwarding.saleKey}
         />

       <input
         id="save"
         type="submit"
         disabled={props.saving}
         value="Save"
         className="btn btn-primary"
         onClick={props.onSubmit}
         />
     </div>
  );
};

export default PointsAwardingForm;
