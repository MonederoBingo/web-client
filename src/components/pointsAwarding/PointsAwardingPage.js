import React from 'react';
import PointsAwardingForm from './children/PointsAwardingForm';

class PointsAwardingPage extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      pointsAwarding: {},
      saving: false,
    };
  }
  updateFieldsState() {

  }
  submit() {

  }
   render() {
     return (
       <div>
         <PointsAwardingForm
           onChange={this.updateFieldsState}
           onSubmit={this.submit}
           pointsAwarding={this.state.pointsAwarding}
           saving={this.state.saving}
         />
       </div>
     );
   }
}

export default PointsAwardingPage;
