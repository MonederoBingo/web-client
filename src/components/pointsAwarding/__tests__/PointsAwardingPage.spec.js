import React from 'react';
import expect from 'jest-matchers';
import PointsAwardingPage from '../PointsAwardingPage';
import PointsAwardingForm from '../children/PointsAwardingForm';
import { shallow } from 'enzyme';

describe("PointsAwardingPage", () => {
  it("should render initial layout", () => {
    // when
    const wrapper = shallow(<PointsAwardingPage/>);

    // then
    expect(wrapper.find('div').children().nodes).toMatchSnapshot();
  });
  it("should call updateFieldsState method in form", () => {
    // given
    jest.spyOn(PointsAwardingPage.prototype, 'updateFieldsState');
    const form = shallow(<PointsAwardingPage />).find('PointsAwardingForm');

    // when
    form.props().onChange();

    // then
    expect(PointsAwardingPage.prototype.updateFieldsState).toHaveBeenCalled();
  });
  it("should pass 'saving' from state to form", () => {
    // given
    jest.spyOn(PointsAwardingPage.prototype, 'updateFieldsState');
    const component = shallow(<PointsAwardingPage />);

    // when
    component.setState({saving: true});

    // then
    expect(component.find('PointsAwardingForm').props().saving).toBe(true);
  });
  it("should call submit method in form", () => {
    // given
    jest.spyOn(PointsAwardingPage.prototype, 'submit');
    const form = shallow(<PointsAwardingPage />).find('PointsAwardingForm');

    // when
    form.props().onSubmit();

    // then
    expect(PointsAwardingPage.prototype.submit).toHaveBeenCalled();
  });
});
