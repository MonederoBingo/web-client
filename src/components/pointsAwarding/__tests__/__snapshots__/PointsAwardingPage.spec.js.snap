// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsAwardingPage should render initial layout 1`] = `
Array [
  <PointsAwardingForm
    onChange={[Function]}
    onSubmit={[Function]}
    pointsAwarding={Object {}}
    saving={false}
  />,
]
`;
